### ISSUE TITLE
Make CA1305 more intelligent (or fix the API)

### ISSUE THREAD

**Issue by pgrawehr** [dotnet/roslyn-analyzers#2481]
Rule CA1305 prevents that I can use the $-Syntax for string concatenation when I want to use the current culture. There's no `FormattableString.CurrentCulture()` method. In addition, the rule has false positives if the argument type is not language dependent (such as a string).

> #### Document Details
> 

#### Analyzer package

Example: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.2

#### Diagnostic ID

CA1305

#### Repro steps

Use $-Syntax to create a string in the current culture:

var error = $"Could not find file {fileName}."; // this isn't even language dependent
var msg = $"Please wait... {percent:F1}% done";  // I don't want to use String.Format with this.

#### Expected behavior
Either:
- Don't report CA1305 if the argument(s) are not language dependent (string, int) or 
- Don't report CA1305 with $-type strings or 
- Add FormattableString.CurrentCulture() method or
- ??
#### Actual behavior
- The above examples generate CA1305, which is only solvable by using traditional String.Format(CultureInfo.CurrentCulture, ...) syntax

**Comment by Evangelink** [comment]
I guess the rule implementation was update in between. The following test-code doesn't raise any diagnostic:

```csharp
[Fact]
public async Task CA1305_StringInterpolation_NoDiagnostic()
{
	await VerifyCS.VerifyAnalyzerAsync(@"
public class C
{
	public C(string fileName, float percent)
	{
		var error = $""Could not find file {fileName}."";
		var msg = $""Please wait... {percent:F1}% done"";
	}
}");
}
```

cc @mavasani

**Comment by Bouke** [comment]
There's #2938 that should solve some false positives. However I'd really like to also see support for interpolated strings. So in your example @Evangelink, there should be errors reported for the implicit `float.ToString()`.

**Comment by pgrawehr** [comment]
I would actually rather see a `FormattableString.CurrentCulture()` implementation, or some easy way of telling the compiler that "yes, I'm aware that this is culture specific and I want it to be!"
